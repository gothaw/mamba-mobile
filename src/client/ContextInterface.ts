export interface ContextInterface {
  container: object
}