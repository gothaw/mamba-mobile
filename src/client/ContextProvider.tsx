import React, { createContext, FunctionComponent } from 'react';

import { ContextInterface } from "../types";
import { theme } from "../theme";

type Props = {
  children: React.ReactNode
};

const Context = createContext<ContextInterface | null>(null);

const ContextProvider: FunctionComponent<Props> = ({children}) => {
  return (
    <Context.Provider value={theme}>
      {children}
    </Context.Provider>
  );
};

export { Context, ContextProvider };
