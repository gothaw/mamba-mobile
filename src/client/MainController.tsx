import { createContext } from "react";
import { Route, NativeRouter, Routes } from "react-router-native";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

const MainController = () => {
  const Context = createContext({});

  return (
    <Context.Provider value={{}}>
      <NativeRouter>
        <Routes>
          <Route path="/" element={<MainMenu/>}/>
          <Route path="/game" element={<Game/>}/>
        </Routes>
      </NativeRouter>
    </Context.Provider>
  );
}

export default MainController;