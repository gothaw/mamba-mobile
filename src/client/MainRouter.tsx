import React, { FunctionComponent, useContext } from "react";
import { View } from "react-native";
import { NativeRouter, Route, Routes } from "react-router-native";

import Icon from "../components/Icon";
import { Images, Paths } from "../config";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

import { Context } from "./ContextProvider";

interface Props {
}

const MainRouter: FunctionComponent<Props> = () => {

  const context = useContext(Context);

  return (
    <View style={{...context.theme.container}}>
      <NativeRouter>
        <Routes>
          <Route path={Paths.MainMenu} element={<MainMenu/>}/>
          <Route path={Paths.Game} element={<Game/>}/>
        </Routes>
        <Icon
          src={Images.MenuIcon}
          style={context.theme.mainMenuIconWrapper}
          imgStyle={context.theme.mainMenuIconImg}
        />
      </NativeRouter>
    </View>
  );
};

export default MainRouter;
