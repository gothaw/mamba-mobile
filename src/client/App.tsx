import MainController from "./MainController";

const App = () => {
  return (
    <MainController/>
  );
}

export default App;