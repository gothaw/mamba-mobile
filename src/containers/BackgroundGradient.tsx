import React, { FunctionComponent, useContext } from "react";
import { LinearGradient } from "expo-linear-gradient";

import { Context } from "../client/ContextProvider";
import { Palette } from "../theme";

type Props = {
  children: React.ReactNode
};


const BackgroundGradient: FunctionComponent<Props> = (props) => {

  const { children } = props;
  const context = useContext(Context);

  return (
    <LinearGradient
      style={{...context.theme.container}}
      colors={[`${Palette.BackgroundLightGreen}`, `${Palette.BackgroundGreen}`]}
      start={{ x: 0, y: 0}}
      end={{ x: 0.25, y: 1.0}}
    >
      {children}
    </LinearGradient>
  );
};

export default BackgroundGradient;
