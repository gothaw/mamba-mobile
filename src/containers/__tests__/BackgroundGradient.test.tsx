import { Text } from "react-native";
import { act, render, waitFor } from "@testing-library/react-native";

import { ContextProvider } from "../../client/ContextProvider";
import BackgroundGradient from "../BackgroundGradient";

describe("BackgroundGradient", () => {
  it("should render without crashing", async () => {
    const gradient = render(
      <ContextProvider>
        <BackgroundGradient>
          <Text>Test</Text>
        </BackgroundGradient>
      </ContextProvider>
    );

    await act(async () => {
      await new Promise(resolve => setImmediate(resolve));
    });

    await waitFor(() => {
      expect(gradient.toJSON()).toMatchSnapshot();
    });
  });
});