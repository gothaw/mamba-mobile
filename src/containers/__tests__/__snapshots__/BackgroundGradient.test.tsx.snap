// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundGradient should render without crashing 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4290491705,
        4287202352,
      ],
      "endPoint": [
        0.25,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        0,
        0,
      ],
    }
  }
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <Text>
    Test
  </Text>
</ViewManagerAdapter_ExpoLinearGradient>
`;
